src/map/utils/battleutils.cpp

```diff
@@ -1547,6 +1547,12 @@ namespace battleutils
         hitrate = hitrate + (acc - eva) / 2 + (PAttacker->GetMLevel() - PDefender->GetMLevel()) * 2;
 
         uint8 finalhitrate = std::clamp(hitrate, 20, 95);
+
+        // Check to see if distance is greater than 25 and force hitrate to be 0
+        if (distance(PAttacker->loc.p, PDefender->loc.p) > 25)
+        {
+            finalhitrate = 0;
+        }
```

Comment:
This is an exception, which means that it could be placed at the very top of the function and avoid having to do any calculations or additional cecks for an attack thats gonna miss always.
```c++
        if (distance(PAttacker->loc.p, PDefender->loc.p) > 25)
        {
            return 0;
        }
```

Reply:
Is this preferred? I'm not well versed in C. It appears I need to do some work anyways since the checks are failing so I might just do a separate PR for this.